src/scss/foundations/colors.scss

```diff
@@ -0,0 +1,119 @@
+@use "sass:color";
+@use "../lib";
+
+$_colorActionBase: #067f96;
+$_colorActionDark: #e5f1f4;
+$_colorActionLight: #045767;
+
+$_colorAlertBase: #ffdc50;
+$_colorAlertDark: #fff8dc;
+$_colorAlertLight: #ffc010;
+
+$_colorBrandBase: #8334c2;
+$_colorBrandDark: #f2eaf8;
+$_colorBrandLight: #5b2488;
+
+$_colorErrorBase: #df1654;
+$_colorErrorDark: #fbe7ed;
+$_colorErrorLight: #b21143;
+
+$_colorNeutral25: #595d6a;
+$_colorNeutral50: #90939d;
+$_colorNeutral75: #c7c9cd;
+$_colorNeutral85: #dddee1;
+$_colorNeutral95: #f3f4f4;
+$_colorNeutralBase: #21283b;
+$_colorNeutralLight: #ffffff;
+
+$_colorSuccessBase: #1d8147;
+$_colorSuccessDark: #e8f2ec;
+$_colorSuccessLight: #145a31;
+
+/*
+The color palette of Pyxis
+
+@type Map
+
+@access private
+*/
+
+$_colors: (
+  action: (
+    base: $_colorActionBase,
+    dark: $_colorActionDark,
+    light: $_colorActionLight,
+  ),
+  alert: (
+    base: $_colorAlertBase,
+    dark: $_colorAlertDark,
+    light: $_colorAlertLight,
+  ),
+  brand: (
+    base: $_colorBrandBase,
+    dark: $_colorBrandDark,
+    light: $_colorBrandLight,
+  ),
+  error: (
+    base: $_colorErrorBase,
+    dark: $_colorErrorDark,
+    light: $_colorErrorLight,
+  ),
+  neutral: (
+    25: $_colorNeutral25,
+    50: $_colorNeutral50,
+    75: $_colorNeutral75,
+    85: $_colorNeutral85,
+    95: $_colorNeutral95,
+    base: $_colorNeutralBase,
+    light: $_colorNeutralLight,
+  ),
+  success: (
+    base: $_colorSuccessBase,
+    dark: $_colorSuccessDark,
+    light: $_colorSuccessLight,
+  ),
+);
+
+/*
+Returns the $color hex with a $variant from map $_colors
+
+@param {String} $color - Name of color
+@param {String} $variant - Name of variant associated with $color, by default is "base"
+
+@return {Color} Representation of named color
+@throws {Error}
+
+@access public
+*/
+
+@function color($color, $variant: base) {
+  @if not map-has-key($_colors, $color) {
+    @error lib.keyNotFoundError("$_colors", $color);
+  }
+
+  @if not map-has-key($_colors, $color, $variant){
+    @error lib.keyNotFoundError("$_colors", $color, $variant);
+  }
+
+  @return map-get($_colors, $color, $variant);
+}
+
+/*
+Create atom classes for all colors variant in $_colors map.
+
+@access public
+*/
+
+@mixin createAtoms() {
+  @each $color, $tones in $_colors {
+    @each $colorTone, $colorHex in $tones {
+      .bg-#{lib.camelToKebabCase($color)}-#{$colorTone} {
+        background-color: $colorHex;
+      }
+
+      .c-#{lib.camelToKebabCase($color)}-#{$colorTone} {
+        color: $colorHex;
+      }
+    }
+  }
+}
```

Comment:
Anche questo mixin lo vedo un po dannoso perché da per scontato che è possibile utilizzare qualsiasi combinazione di colore e background combinando le varie classi.

Reply:
A prescindere dal discorso di combinazione di colore e background (che comunque deve partire dal designer, che deve rispettare delle regole di utilizzo e accessibilità e passare delle review in tal senso), sapete bene che non sono un grande fun delle utilitty-classes, sopratutto quella per i background mi sembra veramente poco utile. Le ho lasciate entrambe in quanto molti si trovano bene ad usarle, ma parliamone pure. Per me una soluzione intermedia potrebbe essere quella di lasciare solo le utility per il colore del testo, così da invogliare l'utilizzatore a usare il css.